sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -74,6 +76,7 @@ class DataSourceV2SQLSuite
 
   before {
     spark.conf.set("spark.sql.catalog.testcat", classOf[InMemoryTableCatalog].getName)
+    spark.conf.set("spark.sql.catalog.testpart", classOf[InMemoryPartitionTableCatalog].getName)
```

Comment:
This suite is growing too big. Can we create a `AlterTablePartitionV2Suite` to put the new tests? We can also create a base trait for v2 tests, so that it can be shared between `DataSourceV2SQLSuite` and the new `AlterTablePartitionV2Suite`.